src/dist.h

```diff
@@ -0,0 +1,15 @@
+/*
+  Copyright (c) 2024, Intel Corporation
+
+  SPDX-License-Identifier: BSD-3-Clause
+*/
+
+/** @file dist.h
+    @brief Defines function with distribution specific implementations.
```

Comment:
If you want to define a common interface here why not to create a base class and inherit slim and composite from it?

Reply:
What issue/problem does it solve?

Reply:
I didn't notice that only one of them can be used in compile time. It's OK then.